src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -29,6 +31,13 @@ def branch_package(workflow_filters = {})
   def check_source_access
     return if remote_source?
 
+    # we don't have any package records on the frontend level for scmsynced projects, therefore
+    # we can only check on the project level for sourceaccess permission
+    if scm_synced_project?
+      Pundit.authorize(@token.user, Project.get_by_name(source_project_name), :source_access?)
```

Comment:
```suggestion
      Pundit.authorize(@token.executor, Project.get_by_name(source_project_name), :source_access?)
```

Reply:
thanks, overlooked the change when I rebased with master